provisioner/clusterpy.go

```diff
@@ -621,47 +622,50 @@ func (p *clusterpyProvisioner) prepareProvision(logger *log.Entry, cluster *api.
 		PollInterval:                   p.updateStrategy.PollInterval,
 	}
 
+	client, err := kubernetes.NewKubeClientWithTokenSource(cluster.APIServerURL, p.tokenSource)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// setup updater
+
 	// allow clusters to override their drain settings
-	handleDurationItem(cluster.ConfigItems, "drain_grace_period", func(v time.Duration) { drainConfig.ForceEvictionGracePeriod = v })
-	handleDurationItem(cluster.ConfigItems, "drain_min_pod_lifetime", func(v time.Duration) { drainConfig.MinPodLifetime = v })
-	handleDurationItem(cluster.ConfigItems, "drain_min_healthy_sibling_lifetime", func(v time.Duration) { drainConfig.MinHealthyPDBSiblingLifetime = v })
-	handleDurationItem(cluster.ConfigItems, "drain_min_unhealthy_sibling_lifetime", func(v time.Duration) { drainConfig.MinUnhealthyPDBSiblingLifetime = v })
-	handleDurationItem(cluster.ConfigItems, "drain_force_evict_interval", func(v time.Duration) { drainConfig.ForceEvictionInterval = v })
-	handleDurationItem(cluster.ConfigItems, "drain_poll_interval", func(v time.Duration) { drainConfig.PollInterval = v })
+	for _, setting := range []struct {
+		key string
+		fn  func(duration time.Duration)
+	}{
+		{"drain_grace_period", func(v time.Duration) { drainConfig.ForceEvictionGracePeriod = v }},
+		{"drain_min_pod_lifetime", func(v time.Duration) { drainConfig.MinPodLifetime = v }},
+		{"drain_min_healthy_sibling_lifetime", func(v time.Duration) { drainConfig.MinHealthyPDBSiblingLifetime = v }},
+		{"drain_min_unhealthy_sibling_lifetime", func(v time.Duration) { drainConfig.MinUnhealthyPDBSiblingLifetime = v }},
+		{"drain_force_evict_interval", func(v time.Duration) { drainConfig.ForceEvictionInterval = v }},
+		{"drain_poll_interval", func(v time.Duration) { drainConfig.PollInterval = v }},
+	} {
+		if value, ok := cluster.ConfigItems[setting.key]; ok {
+			parsed, err := time.ParseDuration(value)
+			if err != nil {
+				return nil, nil, nil, fmt.Errorf("invalid value for %s: %v", setting.key, err)
+			}
+			setting.fn(parsed)
+		}
+	}
```

Comment:
I'm not sure this change makes it easier to read the code?

Reply:
The change makes it actually handle the errors, previously it ignored those completely. :(